BeardedSpice/SaavnStrategy.m

```diff
@@ -0,0 +1,52 @@
+//
+//  SaavnStrategy.m
+//  BeardedSpice
+//
+//  Created by Yash Aggarwal on 1/6/15.
+//  Copyright (c) 2015 Tyler Rhodes / Jose Falcon. All rights reserved.
+//
+
+#import "SaavnStrategy.h"
+
+@implementation SaavnStrategy
+
+-(id) init
+{
+    self = [super init];
+    if (self) {
+        predicate = [NSPredicate predicateWithFormat:@"SELF LIKE[c] '*saavn.com*'"];
+    }
+    return self;
+}
+
+-(BOOL) accepts:(id <Tab>)tab
+{
+    return [predicate evaluateWithObject:[tab URL]];
+}
+
+-(NSString *) toggle
+{
+    return @"(function(){var e=document.querySelectorAll('#play')[0];var t=document.querySelectorAll('#pause')[0];if(t.className.indexOf('hide')===-1){t.click()}else{e.click()}})()";
+}
+
+-(NSString *) previous
+{
+    return @"(function(){var e=document.querySelectorAll('#rew')[0];e.click()})()";
```

Comment:
Why create a variable at all? That’s unnecessary work: it could just be `document.querySelectorAll('#rew')[0].click()`. But even that would be highly inefficient here. If you’re going to select the first element anyway, you don’t need `querySelectorAll`, you can simply use `querySelector` (i.e. `document.querySelector('#rew').click()`), but you’re clicking IDs, and there can be only one per page anyway, so the best way to go about it would be just using `getElementById`: `document.getElementById('rew').click()`.